### ISSUE TITLE
OneNote import

### ISSUE THREAD

**Issue by flxzt** [flxzt/rnote#744]
**Is your feature request related to a problem? Please describe.**  

The only import option that keeps notes in a "rich" and editable way is the xournal++ file import. Because many potential users come from using OneNote, we should consider adding a direct import & conversion from their `.one` files. It's a bit hidden in OneNote, but users are able to save them from onenote or download them from OneDrive when saved in the cloud.

**Describe the solution you'd like**  

Another import option, similar to the Xournal++ import.

**Additional context**  

also discussed in #733 .

I found some rust crates for parsing/processing onenote files: [onenote.rs](https://github.com/msiemens/onenote.rs) is a parser, but currently only supports the format that is saved in OneNote's cloud. [one2html](https://github.com/msiemens/one2html) and [one_note_to_xopp](https://github.com/jonas-kell/one_note_to_xopp) are crates that utilize this parser and save to HTML / .xopp format.

**Comment by einsJannis** [comment]
is this already beeing worked on?

**Comment by flxzt** [comment]
I have a WIP branch locally, but it is stuck because of a [bug](https://github.com/msiemens/onenote.rs/issues/11) in the parsing library that I wanted to use.

**Comment by justinrubek** [comment]
@flxzt do you still have your branch somewhere? The bug mentioned looks like it could be a serious issue, but also never mentions stroke data directly which is what I am interested in because I have a number of OneNote notebooks that I primarily want the handwritten notes and drawings from. I'd be willing to make an attempt to look into this given that importing at least those things may be possible and I may be able to work on that bug as part of it.

**Comment by Doublonmousse** [comment]
One thing to note : All microsoft-based applications (onenote, microsoft journal but also office) copy the InkML information of strokes when selecting ink strokes and copying them to the clipboard as well as Ink Serialized Format.
![image](https://github.com/flxzt/rnote/assets/115779707/3643380e-1c7f-4653-97ab-9a5387839093)

When image are selected, base64 encoding of image seems to be available under the HTML format.

InkML is an open W3C format that's readable (contrary to ISF which is a serialized format, although converter exist and the standard is open as well).
Unfortunately, this info is only available when selecting strokes and copying it to the clipboard, and not in any export menu.

Anyway, I think supporting InkML as an import format (when pasting from the clipboard) could be done and with a little more work 1-1 import of *selected* parts of onenote copied to the clipboard could be achieved in rnote.

Adding InkML as a format when copying stroke from rnote could be done as well and would allow for strokes coming from rnote to be imported as ink strokes in office and onenote applications.

**Comment by GuiFlam** [comment]
Any progress on this?

**Comment by Doublonmousse** [comment]
I've only started to look into parsing/writing inkML files/clipboard data very recently. Can't make promises on how much time it will take until I can push a PR for this though.

There's some weirdness in how onenote writes/parses inkml files (it's definitely very far from implementing the full spec, some values are hard-coded, especially for scaling, it'll always be a 1 int difference <-> 1/1000 of a cm even if you explicitly write another unit and resolution values). And I'm not sure I can match the variable stroke width between rnote and onenote. 

In the inkml file, there's a base width/height for each stroke (per brush) + a pressure channel but I'm pretty sure matching the brush size in rnote and reading the pressure value will give the same result (for one thing, I don't know how onenote does the pressure to thickness mapping, it's definitely not what rnote is doing though).

**Comment by einsJannis** [comment]
> I have a WIP branch locally, but it is stuck because of a [bug](https://github.com/msiemens/onenote.rs/issues/11) in the parsing library that I wanted to use.

In an surprising turn of events, this issue has been resolved after 3 years of radio silence.

**Comment by Doublonmousse** [comment]
Yeah, maybe it'd be worth waiting for https://github.com/msiemens/onenote.rs/issues/19 so the newly added OneNote 2016 `.one` format is added (joplin used a fork and added a few things to it). (I think we're limited to OneNote's cloud format only on the released version for now)
